src/database/migrations/0055-create-audit-log.sql

```diff
@@ -0,0 +1,439 @@
+ qualified if not on search_path audit_rows: Record each row change, or only
+ audit at a statement level audit_query_text: Record the text of the client
+ query that triggered the audit event ? audit_inserts: Audit insert statements
+ or only updates / deletes / truncates ? ignored_cols: Columns to exclude from
+ update diffs, ignore updates that change only ignored cols.$body$;
+
+-- Provide a convenience call wrapper for the simplest case
+-- of row-level logging with no excluded cols and query logging enabled.
+CREATE OR REPLACE FUNCTION audit_table(
+	target_table regclass
+) RETURNS void AS $body$
+SELECT audit_table($1, BOOLEAN 't', BOOLEAN 't', BOOLEAN 't', ARRAY []::text []);
+$body$ LANGUAGE sql;
+COMMENT ON FUNCTION audit_table(regclass) IS $body$
+Add auditing support to the given table.Row - level changes will be logged with
+ full client query text.No cols are ignored.$body$;
+CREATE OR REPLACE FUNCTION deaudit_table(
+	target_table regclass
+) RETURNS void AS $body$ BEGIN
+-- Log the deaudit call event into the audit table.
+INSERT INTO audit_logs (
+	event_id,
+	schema_name,
+	table_name,
+	relid,
+	session_user_name,
+	action_tstamp_tx,
+	action_tstamp_stm,
+	action_tstamp_clk,
+	transaction_id,
+	application_name,
+	client_addr,
+	client_port,
+	client_query,
+	action,
+	statement_only)
+SELECT
+	nextval('audit_logs_event_id_seq'),
+	-- event_id
+	relnamespace::regnamespace::text,
+	-- schema_name from pg_catalog.pg_class
+	relname::text,
+	-- table_name from pg_catalog.pg_class
+	target_table::regclass,
+	-- relation OID
+	session_user::text,
+	-- session_user_name
+	current_timestamp,
+	-- action_tstamp_tx
+	statement_timestamp(),
+	-- action_tstamp_stm
+	clock_timestamp(),
+	-- action_tstamp_clk
+	txid_current(),
+	-- transaction ID
+	current_setting('application_name'),
+	-- client application
+	inet_client_addr(),
+	-- client_addr
+	inet_client_port(),
+	-- client_port
+	current_query(),
+	-- top-level query or queries (if multistatement) from client
+	'S',
+	-- action
+	'f' -- statement_only
+FROM pg_catalog.pg_class
+WHERE oid = target_table::regclass;
+EXECUTE 'DROP TRIGGER IF EXISTS audit_trigger_row ON ' || target_table;
+EXECUTE 'DROP TRIGGER IF EXISTS audit_trigger_stm ON ' || target_table;
+END;
+$body$ LANGUAGE plpgsql;
+COMMENT ON FUNCTION deaudit_table(
+	regclass
+) IS $body$ Remove auditing support to the given table.$body$;
+CREATE OR REPLACE VIEW audited_tables AS
+SELECT DISTINCT
+	triggers.trigger_schema AS schema,
+	triggers.event_object_table AS auditedtable
+FROM information_schema.triggers
+WHERE
+	triggers.trigger_name::text IN (
+		'audit_trigger_row'::text,
+		'audit_trigger_stm'::text
+	)
+ORDER BY
+	schema,
+	auditedtable;
+COMMENT ON VIEW audited_tables IS $body$ View showing all tables with auditing
+set up.Ordered by schema,
+	then table.$body$;
+
+-- Enable audit logs on PDC tables
+SELECT
+	audit_table('application_form_fields') AS aff,
```

Comment:
do we need the `AS foo`s here (the output of this select isn't used, right?)

Reply:
I think this was a SQLfluff requirement. Let me check.

Reply:
🤦🏻 @ sqlfluff

Reply:
Yes, blame SQLfluff. I tried to remove one of them and got this:
```
service$ npm run lint:sqlfluff

> lint:sqlfluff
> ./venv/bin/sqlfluff lint src/database/

== [src/database/migrations/0055-create-audit-log.sql] FAIL                                  
L: 424 | P:   2 | AL03 | Column expression without alias. Use explicit `AS`                  
                       | clause. [aliasing.expression]
All Finished 📜 🎉!                                                                          
service$ echo $?
1
```

That's, what, strike 8 against SQLfluff?

Reply:
I think just leaving as is (with aliases) is fine for now (alternatively you could disable the rule via a comment). I was just curious the reason!

Reply:
I tried several test runs (of sources.int.ts) in a row with and without the independent statements and I cannot find a substantial performance difference. Both take about 35 to 38 seconds on my machine. Seems long for both. Perhaps I haven't turned off synchronous commit or something.